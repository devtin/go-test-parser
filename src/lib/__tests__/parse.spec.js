import { readFileSync } from 'fs'
import path from 'path'
import {
    parse,
    getStringIndent,
    unIndentString,
    getFnBlockedContent,
} from '../parse'

const goTestFileFixture = readFileSync(path.join(__dirname, './fixtures/some_test.go')).toString()
const goTestConveyFixture = readFileSync(path.join(__dirname, './fixtures/convey_test.go')).toString()

describe('getStringIndent()', () => {
    it('returns the minimum amount of indentation found in given text', () => {
        const indent = getStringIndent(`  abc\n    123\n  do re mi`)
        expect(indent).toEqual(2)
    })
})

describe('unIndentString()', () => {
    it('removes the minimum amount of indentation found in given text', () => {
        const unIndented = unIndentString(`  abc\n    123\n  do re mi`)
        expect(unIndented).toEqual(`abc\n  123\ndo re mi`)
    })
})

describe('getFnBlockedContent()', () => {
    it('retrieves metadata from a particular set of convey functions', () => {
        const parsedTree = getFnBlockedContent(goTestConveyFixture, ['Convey'])
        expect(parsedTree).toMatchSnapshot()
    })
})

describe('parse()', () => {
    it('retrieves metadata from a go test file', () => {
        const parsedTree = parse(goTestFileFixture)
        expect(parsedTree).toMatchSnapshot()
    })
})
