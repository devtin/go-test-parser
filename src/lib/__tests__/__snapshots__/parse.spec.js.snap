// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFnBlockedContent() retrieves metadata from a particular set of convey functions 1`] = `
Object {
  "Convey": Array [
    Object {
      "Convey": Array [
        Object {
          "Convey": Array [],
          "code": "// some
// test
// code
// block",
          "end": 6,
          "start": 1,
          "title": "Test A-1",
        },
        Object {
          "Convey": Array [],
          "code": "// some
// test
// code
// block",
          "end": 12,
          "start": 7,
          "title": "Test A-2",
        },
      ],
      "code": "Convey(\\"Test A-1\\", func() {
	// some
	// test
	// code
	// block
})
Convey(\\"Test A-2\\", func() {
	// some
	// test
	// code
	// block
})",
      "end": 14,
      "start": 1,
      "title": "Test A",
    },
    Object {
      "Convey": Array [
        Object {
          "Convey": Array [],
          "code": "// some
// test
// code
// block",
          "end": 6,
          "start": 1,
          "title": "Test B-1",
        },
        Object {
          "Convey": Array [],
          "code": "// some
// test
// code
// block",
          "end": 12,
          "start": 7,
          "title": "Test B-2",
        },
      ],
      "code": "Convey(\\"Test B-1\\", func() {
	// some
	// test
	// code
	// block
})
Convey(\\"Test B-2\\", func() {
	// some
	// test
	// code
	// block
})",
      "end": 29,
      "start": 16,
      "title": "Test B",
    },
  ],
}
`;

exports[`parse() retrieves metadata from a go test file 1`] = `
Array [
  Object {
    "Convey": Array [
      Object {
        "Convey": Array [
          Object {
            "Convey": Array [],
            "code": "// some
// test
// code
// block",
            "end": 6,
            "start": 1,
            "title": "Test A-1",
          },
          Object {
            "Convey": Array [],
            "code": "// some
// test
// code
// block",
            "end": 12,
            "start": 7,
            "title": "Test A-2",
          },
        ],
        "code": "Convey(\\"Test A-1\\", func() {
	// some
	// test
	// code
	// block
})
Convey(\\"Test A-2\\", func() {
	// some
	// test
	// code
	// block
})",
        "end": 16,
        "start": 3,
        "title": "Test A",
      },
      Object {
        "Convey": Array [
          Object {
            "Convey": Array [],
            "code": "// some
// test
// code
// block",
            "end": 6,
            "start": 1,
            "title": "Test B-1",
          },
          Object {
            "Convey": Array [],
            "code": "// some
// test
// code
// block",
            "end": 12,
            "start": 7,
            "title": "Test B-2",
          },
        ],
        "code": "Convey(\\"Test B-1\\", func() {
	// some
	// test
	// code
	// block
})
Convey(\\"Test B-2\\", func() {
	// some
	// test
	// code
	// block
})",
        "end": 31,
        "start": 18,
        "title": "Test B",
      },
    ],
    "code": "// some setup

Convey(\\"Test A\\", t, func() {
	Convey(\\"Test A-1\\", func() {
		// some
		// test
		// code
		// block
	})
	Convey(\\"Test A-2\\", func() {
		// some
		// test
		// code
		// block
	})
})

Convey(\\"Test B\\", t, func() {
	Convey(\\"Test B-1\\", func() {
		// some
		// test
		// code
		// block
	})
	Convey(\\"Test B-2\\", func() {
		// some
		// test
		// code
		// block
	})
})",
    "end": 44,
    "start": 12,
    "title": "Some",
  },
]
`;
